libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -188,6 +199,34 @@ size_t ContractCompiler::packIntoContractCreator(ContractDefinition const& _cont
 	return m_context.runtimeSub();
 }
 
+size_t ContractCompiler::deployLibrary(ContractDefinition const& _contract)
+{
+	solAssert(!!m_runtimeCompiler, "");
+	solAssert(_contract.isLibrary(), "Tried to deploy contract as library.");
+
+	CompilerContext::LocationSetter locationSetter(m_context, _contract);
+
+	solAssert(m_context.runtimeSub() != size_t(-1), "Runtime sub not registered");
+	m_context.pushSubroutineSize(m_context.runtimeSub());
+	m_context.pushSubroutineOffset(m_context.runtimeSub());
+	m_context.appendInlineAssembly(R"(
+	{
+		// If code starts at 11, an mstore(0) writes to the full PUSH20 plus data
+		// without the need for a shift.
```

Comment:
If `PushDeployTimeAddress` uses `PUSH32` then a single `MSTORE` could be used here.

Reply:
This saves up to 2400 gas at deploy time.

Reply:
The difference in bytecode size (12 bytes) should cost an extra `68 * 12 = 816`.

However by removing the need for the switch statement should reduce instruction costs.

Reply:
These 12 bytes are stored in the contract code, where the cost is 200 gas per byte.